src/k8s/cmd/k8s/k8s_disable.go

```diff
@@ -9,36 +10,60 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var (
-	disableCmd = &cobra.Command{
-		Use:       "disable <component>",
-		Short:     "Disable a specific component in the cluster",
-		Long:      fmt.Sprintf("Disable one of the specific components: %s.", strings.Join(componentList, ",")),
-		Args:      cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
-		ValidArgs: componentList,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			name := args[0]
-
-			client, err := client.NewClient(cmd.Context(), client.ClusterOpts{
-				StorageDir: clusterCmdOpts.storageDir,
-				Verbose:    rootCmdOpts.logVerbose,
-				Debug:      rootCmdOpts.logDebug,
-			})
-			if err != nil {
-				return fmt.Errorf("failed to create client: %w", err)
-			}
-
-			err = client.UpdateComponent(cmd.Context(), name, api.ComponentDisable)
-			if err != nil {
-				return fmt.Errorf("failed to %s %s: %w", name, api.ComponentDisable, err)
-			}
-
-			fmt.Printf("Component %q disabled.\n", name)
-			return nil
-		},
-	}
-)
+type disableFunc func(ctx context.Context, client *client.Client) error
+
+var disableActions = map[string]disableFunc{
+	"dns":     disableDns,
+	"network": disableNetwork,
+}
+
+var disableCmd = &cobra.Command{
```

Comment:
This looks great, but I would recommend having separate commands here as well. It is much easier to be able to specify custom arguments in the future (if we have to), same as we have in the enable command.